update-workflow

```diff
@@ -31,7 +33,7 @@ main() {
   fi
 
   echo "Add or update common Github workflow files."
-  cp "$config_root/files/workflows/"* "$enrolling_root/.github/workflows"
+  ls "$config_root/files/workflows" | xargs -L1 -I% bash -c "$config_root/files/workflows/% > $enrolling_root/.github/workflows/%"
```

Comment:
Slightly simpler / more straightforward alternative:
```bash
for file in "$config_root"/files/workflows/*; do
  "$config_root/files/workflows/$file" >
    "$enrolling_root/.github/workflows/$file"
done
```

Reply:
I'm very much opposed to having files with the extension be `.yml`, but really be bash scripts, and require them to be executable.  It's a surprising behavior and non-obvious.  I would much rather the files be renamed and handled appropriately (as in, `git_rules.yml` becomes `git_rules.yml.sh` and it executes the file and writes the output to `git_rules.yml`), or my preference the files changed back to YAML files and evaluated as templates to replace variables.

Something like the following could work (this example would require double quotes in the file to be escaped)
```bash
files=$(ls my/dir)

for filename in $files
do
  in_file="my/dir/${filename}"
  out_file="out/dir/${filename}"
  eval "cat <<EOF > '${out_file}'
$(<${in_file})
EOF"
done
```

Reply:
I 💯 agree with this comment, and should have caught that too.

But re: bash snippet, use a glob, not ls, to loop through the files:

http://mywiki.wooledge.org/BashPitfalls#for_f_in_.24.28ls_.2A.mp3.29

Reply:
Also, I'm not crazy about using `eval` here.  I think I prefer the `.sh` suggestion since that's more explicit about our approach.

Reply:
I'm not a fan of using eval, either.  I'd actually much prefer the yaml files be treated as templates, so a yml file with no vars in it is preserved across and one with vars only expands them into the output, however that's accomplished (the eval was just a top of my head sketch).  Even something as simple as .sh are executed and everything else copied would work.